app/models/previous_teacher_training.rb

```diff
@@ -1,8 +1,33 @@
 class PreviousTeacherTraining < ApplicationRecord
   belongs_to :application_form
+  belongs_to :provider, optional: true
 
   enum :started, {
     yes: 'yes',
     no: 'no',
-  }
+  }, prefix: true
+
+  enum :status, {
+    draft: 'draft',
+    published: 'published',
+  }, default: 'draft'
+
+  def create_draft_dup!
+    dup_record = dup
+    dup_record.status = 'draft'
+
+    dup_record.save!
+    dup_record
+  end
+
+  def reviewable?
+    return true if started_yes? && [provider_name, started_at, details].all?(&:present?)
+    return true if started_no? && [provider_name, started_at, details].all?(&:nil?)
+
+    false
```

Comment:
```suggestion
    (started_yes? && [provider_name, started_at, details].all?(&:present?)) ||
        (started_no? && [provider_name, started_at, details].all?(&:nil?))
```

This is maybe just a personal preference, but I don't really like returning true or false explicitly.